src/analysis/normed_space/affine_isometry.lean

```diff
@@ -587,3 +588,44 @@ begin
   rw this,
   simp only [homeomorph.comp_is_open_map_iff, homeomorph.comp_is_open_map_iff'],
 end
+
+local attribute [instance, nolint fails_quickly] affine_subspace.nonempty_map
+
+include V₁
+omit V
+
+/--
+An affine subspace is isomorphic to its image under an injective affine map.
+This is the affine version of `submodule.equiv_map_of_injective`.
+-/
+noncomputable def affine_map.equiv_map_of_injective
+  (E: affine_subspace 𝕜 P₁) [nonempty E]
+  (φ : P₁ →ᵃ[𝕜] P₂) (hφ : function.injective φ) : E ≃ᵃ[𝕜] E.map φ :=
+affine_equiv.of_bijective (affine_map.restrict.bijective hφ)
+
+namespace affine_isometry
+
+/-- Restricts an affine isometry to an affine isometry equivalence between a nonempty affine
+subspace `E` and its image. -/
```

Comment:
```suggestion
subspace `E` and its image.

This is a stronger version of `affine_subspace.equiv_map`. -/
```

Reply:
Because it's not really a stronger version (the premise is also stronger), I wrote:
```
This is an isometry version of `affine_subspace.equiv_map`, having a stronger premise and a stronger
conclusion.
```
Hope that's fine!